sdk/common/src/main/java/io/opentelemetry/sdk/resources/EntityBuilder.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.resources;
+
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.common.AttributesBuilder;
+import java.util.Objects;
+import java.util.function.Consumer;
+import javax.annotation.Nullable;
+
+/**
+ * A builder of {@link Entity} that allows to add identifying or descriptive {@link Attributes}, as
+ * well as type and schema_url.
+ */
+public class EntityBuilder {
+  @Nullable private String entityType;
+  private final AttributesBuilder attributesBuilder;
+  private final AttributesBuilder identifyingBuilder;
+  @Nullable private String schemaUrl;
+
+  EntityBuilder() {
+    this.attributesBuilder = Attributes.builder();
+    this.identifyingBuilder = Attributes.builder();
+  }
+
+  EntityBuilder(Entity seed) {
+    this.entityType = seed.getType();
+    this.schemaUrl = seed.getSchemaUrl();
+    this.identifyingBuilder = seed.getIdentifyingAttributes().toBuilder();
+    this.attributesBuilder = seed.getAttributes().toBuilder();
+  }
+
+  /**
+   * Assign an OpenTelemetry schema URL to the resulting Entity.
+   *
+   * @param schemaUrl The URL of the OpenTelemetry schema being used to create this Entity.
+   * @return this
+   */
+  public EntityBuilder setSchemaUrl(String schemaUrl) {
+    this.schemaUrl = schemaUrl;
+    return this;
+  }
+
+  /**
+   * Assign an Entity type to the resulting Entity.
+   *
+   * @param entityType The type name of the entity.
+   * @return this
+   */
+  public EntityBuilder setEntityType(String entityType) {
+    this.entityType = entityType;
+    return this;
+  }
+
+  /**
+   * Modify the descriptive attributes of this Entity.
+   *
+   * @param f A thunk which manipulates descriptive attributes.
```

Comment:
What's a thunk?

Separately, what's the motivation behind a consumer rather than the simpler `withDescriptive(Attributes atributes)`? The consumer pattern is more clever and lower overhead, but is less intuitive and this probably isn't a place where efficiency matter.s

Reply:
This was just me taking shortcuts in adding all the various methods.  I think if we submit this we should expand to ALL relevant methods of `add{Identifying|Descriptive}Attribute{s?}`.

A "thunk" is a term in other languages for first-class functions, lambdas, etc.  I can update this API to actually be the one we'd expose.  For now, just assume you have a bunch of reasonable methods to add attributes :).